tests/compiler/test_compile_export.py

```diff
@@ -130,7 +130,7 @@ def test_export_compile_data_to_file_categories(
     report_exnc,
     tempfile_export,
 ) -> None:
-    snippetcompiler.setup_for_snippet(snippet)
+    snippetcompiler.setup_for_snippet(snippet, install_project=not issubclass(exception, ParserException))
```

Comment:
This logic looks a bit weird

Reply:
The implementation or the actual logic? Any suggestions for improvement?